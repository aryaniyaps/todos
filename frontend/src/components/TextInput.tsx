import React, { forwardRef, HTMLProps } from "react";

export interface TextInputInputProps
    extends Omit<HTMLProps<HTMLInputElement>, "prefix" | "size"> {
    invalid?: boolean;
}

export const TextInput = forwardRef<HTMLInputElement, TextInputInputProps>(
    ({ className = "", invalid, ...props }, ref) => {
        const border = invalid ? "border-danger" : "border-primary-800";
        const styles = `w-full py-2 px-3 rounded text-primary-300 placeholder-primary-400 border ${border} transition-colors duration-300 ease-in-out bg-input  ${className}`;

        return <input className={styles} ref={ref} data-testid="input" {...props} />;
    }
);

TextInput.displayName = "TextInput";
