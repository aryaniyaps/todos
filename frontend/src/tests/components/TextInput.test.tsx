import { render } from "@testing-library/react";
import TextInput from "../../components/TextInput";

describe("TextInput tests", () => {
    it("should render a text input", () => {
        const { getByTestId } = render(<TextInput />);
        const textInput = getByTestId("text-input");
        expect(textInput).toBeVisible();
    });

    it("should match snapshot", () => {
        const { getByTestId } = render(<TextInput />);
        const textInput = getByTestId("text-input");
        expect(textInput).toMatchSnapshot();
    });
});
