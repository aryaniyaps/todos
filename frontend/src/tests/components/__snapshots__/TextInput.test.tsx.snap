// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component tests should match snapshot 1`] = `
<input
  class="relative w-full py-2 px-3 shadow-sm rounded-md placeholder-primary-500 border border-primary-400 bg-transparent  "
  data-testid="text-input"
/>
`;
